### PR TITLE
Add support for Custom format validators and remote reference caching

### PR DESCRIPTION
**Pull request by gaspaio**
Support for some very useful but advanced features in Z-Schema.

### CODE REVIEW

**Comment by gaspaio** [comment]
it would be nice to have some feedback on this PR ...

**Comment by petrbela** [comment]
Hey,sorry, I've been pretty busy lately.I'll make sure to check it tomorrow.Thanks for the contribution!

File: tasks/zschema.js [hunk 1]
```diff
@@ -23,6 +23,20 @@ module.exports = function(grunt) {
     // Merge task-specific and/or target-specific options with these defaults.
     var options = this.options({});
 
+    var data = this.data;
+    data.formatValidators = data.validators || {};
```
Comment (petrbela) [inline tasks/zschema.js]: use one name or the other; i see no reason why `validators` should be public api translated to `formatValidators` internally

File: README.md [hunk 2]
```diff
@@ -41,6 +41,40 @@ grunt.initConfig({
...
+    zschema: {
+      validateSchemas: {
+        validators: {
+          'xstring': validators.xstringValidator
+        },
+        remoteRefs: {
+          'http://apis.mysite.com/myschema.json': schemas.myschema,
+        },
+        files: {
+          'schemas/main.json': 'fixtures/**/*.json'
```
Comment (petrbela) [inline README.md:63]: I'm not sure, is this format allowed? I think the value is supposed to be an array.

**Comment by petrbela** [comment]
Can you write some tests for these two features?

File: README.md (hunk 2 above)
Comment (gaspaio) [inline README.md:63]: The xstring format is not a native format in JSON Schema. Z-Schema allows-us to add custom formats : callbacks that validate strings (cf. [documentation](https://github.com/zaggino/z-schema#custom-format-validators)).

**Comment by gaspaio** [comment]
Thanks for the review. Tests + refactoring done !

File: README.md (hunk 2 above)
Comment (petrbela) [inline README.md:63]: I understand the zschema part but not sure you can list files as `'dest': 'src'`, I think it needs to be `'dest': ['src']`. Have you tried it?

**Comment by petrbela** [comment]
Looks good. :+1:

**Pull request merged**